app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -0,0 +1,37 @@
+module OpenidConnect
+  class AuthorizationController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def index
+      @authorize_form = OpenidConnectAuthorizeForm.new(params)
+      if @authorize_form.valid?
+        render :index
+      else
+        analytics.track_event(Analytics::OPENID_CONNECT_MALFORMED,
```

Comment:
What do you think about moving this call up to the top of the method, after assigning the form, so that we can track both the success and failed states in one shot? The hash passed in would be the same as here, but with an additional `success: @authorize_form.valid?`, and the name of the event would be changed to `OPENID_CONNECT_INTIAL_REQUEST` or whatever best describes this action.